Sources/NIOSSL/SSLContext.swift

```diff
@@ -434,15 +434,32 @@ extension NIOSSLContext {
             // This is get0 so we can just ignore the pointer, we don't have an owned ref.
             let trustParams = CNIOBoringSSL_SSL_CTX_get0_param(context)!
             CNIOBoringSSL_X509_VERIFY_PARAM_set_flags(trustParams, CUnsignedLong(X509_V_FLAG_TRUSTED_FIRST))
+            
+            func loadCACertificate(certificate: NIOSSLCertificate) throws {
+                // Adds the CA name extracted from cert to the list of CAs sent to the client when requesting a client certificate.
+                guard 1 == CNIOBoringSSL_SSL_CTX_add_client_CA(context, certificate.ref) else {
+                    throw NIOSSLError.failedToLoadCertificate
+                }
+            }
 
             func configureTrustRoots(trustRoots: NIOSSLTrustRoots) throws {
                 switch trustRoots {
                 case .default:
                     try NIOSSLContext.platformDefaultConfiguration(context: context)
                 case .file(let path):
                     try NIOSSLContext.loadVerifyLocations(path, context: context)
+                    // Add the CA name from the trust root
+                    if sendCANames {
```

Comment:
This code path is not quite right: if you crack open `loadVerifyLocations` you can see that we may have many certificates within the specific location that we might need to handle.

In this instance we probably need to use `loadVerifyLocations` to help us out. If the path is a file, we can use `SSL_load_client_CA_file` to give us a stack of X509 names and use those. If the path is a directory, we have to work a little bit harder, and enumerate the files in the directory and load them.

We'll need test cases for all of these cases as well.

Reply:
Good call out.  I'll get a fix together for this with some tests.

Reply:
Added a new code path here in the latest commit.